### PR TITLE
Support reseting iCx cards

### PR DESCRIPTION
**Pull request by bettse**
# What's new

- Support resetting iClass cards to iCL/iCS/iCE (https://github.com/iCopy-X-Community/icopyx-teardown/blob/master/tags/README.md#icl)

# Verification 

- Scan iClass card with standard key, see new options.
- Choose one of the options
- Use card with iCopyX (successfully)

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by bettse** [comment]
@nvx  / @pcunning Thoughts?

**Comment by nvx** [comment]
Nice. Is it worth adding the iCopy keys to the key search so we can read cards with iCopy keys? (Unless it already does this and I missed the code)

**Comment by pcunning** [comment]
Thanks for implementing my external feedback about moving the menu items into a folder. 

I agree with @nvx that if we are making it this easy to change a key to something it needs to be easy to change back to a standard key. That means we can read cards in a given format they end up in. Biggest issue I see with that is we don’t have a good way to decide what key algorithm we use with the keys. We could just do both or have a way to indicate it in a dictionary file. If we do allow people to input their own key at some point (ui or based on a file) it would be good to make sure that key is also in the user dictionary.

**Comment by bettse** [comment]
So they keys are in the dictionary: https://github.com/flipperdevices/flipperzero-firmware/blob/dev/assets/resources/apps_data/picopass/assets/iclass_elite_dict.txt#L24 but I think it wouldn't find them because they use the standard kdf and not the elite kdf.  Maybe we could run through the system one twice, once elite and once standard?  

I like the idea of having a user input for the key, but that is outside the scope of this to me.

**Comment by pcunning** [comment]
Running through them twice seems good to me.

**Comment by bettse** [comment]
> Running through them twice seems good to me.

Done! 😁

File: applications/plugins/picopass/scenes/picopass_scene_read_factory_success.c [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include "../picopass_i.h"
 #include <dolphin/dolphin.h>
 
+extern const uint8_t picopass_iclass_key[];
```
Comment (skotopes) [inline applications/plugins/picopass/scenes/picopass_scene_read_factory_success.c:4]: The best will be to move this definition to separate .h/.c file

**Review by skotopes — APPROVED** [review]

File: applications/plugins/picopass/scenes/picopass_scene_read_factory_success.c (hunk 1 above)
Comment (bettse) [reply applications/plugins/picopass/scenes/picopass_scene_read_factory_success.c:4]: I'll try to remember to do that in a future PR, but I'm also OK if someone else does, maybe a good contribution for a new person.

Comment (bettse) [reply applications/plugins/picopass/scenes/picopass_scene_read_factory_success.c:4]: I was playing with this, but I don't think I'm familiar with the pattern.  Are their examples in the codebase you can point me to?

Comment (skotopes) [reply applications/plugins/picopass/scenes/picopass_scene_read_factory_success.c:4]: https://github.com/flipperdevices/flipperzero-firmware/blob/0.75.0/firmware/targets/f7/furi_hal/furi_hal_resources.h
https://github.com/flipperdevices/flipperzero-firmware/blob/0.75.0/firmware/targets/f7/furi_hal/furi_hal_resources.c

**Pull request merged**